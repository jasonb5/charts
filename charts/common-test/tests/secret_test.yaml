suite: test secret
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should set secret type
    set:
      secret:
        credentials:
          type: kubernetes.io/dockerconfigjson
          data:
            .dockerconfigjson: '{"auths": {"docker.io": {"email": "user1@domain.io", "auth": {{ printf "%s:%s" "user1" "abcd1234" | b64enc | quote }}}}}'
    asserts:
      - equal:
          path: type
          value: kubernetes.io/dockerconfigjson
      - equal:
          path: data
          value:
            .dockerconfigjson: IntcImF1dGhzXCI6IHtcImRvY2tlci5pb1wiOiB7XCJlbWFpbFwiOiBcInVzZXIxQGRvbWFpbi5pb1wiLCBcImF1dGhcIjogXCJkWE5sY2pFNllXSmpaREV5TXpRPVwifX19Ig==
      - hasDocuments:
          count: 1
  - it: should configure secret
    set:
      secret:
        default:
          data:
            CONFIG: /config/config.txt
            PORT: "{{ .Values.service.ports.default.port }}"

      service:
        ports:
          default:
            port: 80
    asserts:
      - equal:
          path: data
          value:
            CONFIG: Ii9jb25maWcvY29uZmlnLnR4dCI=
            PORT: IjgwIg==
      - hasDocuments:
          count: 1
