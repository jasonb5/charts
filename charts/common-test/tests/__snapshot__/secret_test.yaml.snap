should change type:
  1: |
    apiVersion: v1
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test-credentials
      namespace: NAMESPACE
    type: Opaque
should pass:
  1: |
    apiVersion: v1
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test-config
      namespace: NAMESPACE
    stringData:
      PORT: "80"
    type: Opaque
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      ports:
        - name: common-test
          port: 80
          protocol: TCP
          targetPort: 80
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
